addons/sale_pdf_quote_builder/models/ir_actions_report.py

```diff
@@ -43,108 +52,160 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
-            'validity_date': format_date(env, order.validity_date, lang_code=lang_code),
-            'client_order_ref': order.client_order_ref or '',
+        return {
+            field: self._get_formatted_field(field, order, tz, param_field_map)
+            for field in all_form_fields
         }
 
-        # Adding fields from each line, prefixed by the line_id to avoid conflicts
-        lines_with_doc_ids = set(doc_line_id_mapping.values())
-        for line in order.order_line.filtered(lambda sol: sol.id in lines_with_doc_ids):
-            form_fields_mapping.update(self._get_sol_form_fields_mapping(line))
-
-        return form_fields_mapping
-
-    def _get_sol_form_fields_mapping(self, line):
-        """ Dictionary mapping specific pdf fields name to Odoo fields data for a sale order line.
+    @api.model
+    def _get_formatted_field(self, form_field_name, order, tz, param_field_map):
+        """Format a field value from the extracted PDF string.
 
-        Fields name are prefixed by the line id to avoid conflict between files.
-
-        Override this method to add new fields to the mapping.
-
-        :param recordset line: sale.order.line record
-        :rtype: dict
-        :return: mapping of prefixed fields name to Odoo fields data
+        Note: order.ensure_one()
 
-        Note: line.ensure_one()
+        :param string form_field_name: field path
+        :param recordset order: sale.order record from which to take the value, following the path
+        :param dict translated_env: self.env, with the correct translation context to format amount,
+                                    date and datetime
+        :param str tz: the timezone used for rendering datetime
+        :param dict param_field_map: the map stored in the config parameter to assign the expected
+                                     paths to the form fields. A form field not in that map will be
+                                     ignored.
+        :return: formatted field value
+        :rtype: string
         """
-        line.ensure_one()
-        env = self.with_context(use_babel=True).env
-        return {
-            f'{line.id}_description': line.name,
-            f'{line.id}_quantity': line.product_uom_qty,
-            f'{line.id}_uom': line.product_uom.name,
-            f'{line.id}_price_unit': format_amount(env, line.price_unit, line.currency_id),
-            f'{line.id}_discount': line.discount,
-            f'{line.id}_product_sale_price': format_amount(
-                env, line.product_id.lst_price, line.product_id.currency_id
-            ),
-            f'{line.id}_taxes': ', '.join(tax.name for tax in line.tax_id),
-            f'{line.id}_tax_excl_price': format_amount(env, line.price_subtotal, line.currency_id),
-            f'{line.id}_tax_incl_price': format_amount(env, line.price_total, line.currency_id),
-        }
+        def _get_formatted_value(self, records, field_name):
```

Comment:
```suggestion
        def _get_formatted_value(self, field_name):
```
Not needed (the inner can access a variable from the outer method, as long as it's used in the inner method)